File: backend/catalog-service/src/main/java/me/doruk/catalogservice/request/EventCreateRequest.java
package me.doruk.catalogservice.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.validation.constraints.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class EventCreateRequest {
  @NotBlank
  @Size(max = 255)
  private String name;

  @Positive
  private int totalCapacity;

  @NotNull
  private Long venueId;

  @NotNull
  @DecimalMin(value = "0.0", inclusive = true)
  private BigDecimal ticketPrice;

  @NotNull
  @FutureOrPresent
  private LocalDate eventDate;

  @Size(max = 1000)
  private String description;

  @Size(max = 512)
  private String imageUrl;
}

File: backend/catalog-service/Requests/EventCreateRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace CatalogService.Requests;

public class EventCreateRequest
{
    [Required]
    [StringLength(255)]
    public string Name { get; set; } = default!;

    [Range(1, int.MaxValue)]
    public int TotalCapacity { get; set; }

    [Required]
    public long VenueId { get; set; }

    [Required]
    [Range(typeof(decimal), "0", "79228162514264337593543950335")]
    public decimal TicketPrice { get; set; }

    [Required]
    [DataType(DataType.Date)]
    public DateTime EventDate { get; set; }

    [StringLength(1000)]
    public string? Description { get; set; }

    [StringLength(512)]
    public string? ImageUrl { get; set; }
}
